### ISSUE TITLE
ld.lld-11: error: section ... file range overlaps with ...

### ISSUE THREAD

**Issue by dileks** [ClangBuiltLinux/linux#1129]
With `Linux v5.9-rc1` and `GCC v10.2` and `ld.lld-11` (LLD 11.0.0 (https://github.com/llvm/llvm-project a450654a52874b094c264e0366c31126c03fdf2d) (compatible with GNU linkers))
```
make -f ./scripts/Makefile.build obj=arch/x86/boot/compressed arch/x86/boot/compressed/vmlinux
  ld.lld-11 -m elf_x86_64 --compress-debug-sections=zlib  -m elf_i386 -T arch/x86/boot/setup.ld arch/x86/boot/a20.o arch/x86/boot/bioscall.o arch/x86/boot/cmdline.o arch/x86/boot/copy.o arch/x86/boot/cpu.o arch/x86/boot/cpuflags.o arch/x86/boot/cpucheck.o arch/x86/boot/early_serial_console.o arch/x86/boot/edd.o arch/x86/boot/header.o arch/x86/boot/main.o arch/x86/boot/memory.o arch/x86/boot/pm.o arch/x86/boot/pmjump.o arch/x86/boot/printf.o arch/x86/boot/regs.o arch/x86/boot/string.o arch/x86/boot/tty.o arch/x86/boot/video.o arch/x86/boot/video-mode.o arch/x86/boot/version.o arch/x86/boot/video-vga.o arch/x86/boot/video-vesa.o arch/x86/boot/video-bios.o -o arch/x86/boot/setup.elf
ld.lld-11: error: section .text.startup file range overlaps with .header
>>> .text.startup range is [0x1040, 0x1204]
>>> .header range is [0x11EF, 0x126B]

ld.lld-11: error: section .header file range overlaps with .bsdata
>>> .header range is [0x11EF, 0x126B]
>>> .bsdata range is [0x1205, 0x139E]

ld.lld-11: error: section .bsdata file range overlaps with .entrytext
>>> .bsdata range is [0x1205, 0x139E]
>>> .entrytext range is [0x126C, 0x12D3]

ld.lld-11: error: section .text.startup virtual address range overlaps with .header
>>> .text.startup range is [0x40, 0x204]
>>> .header range is [0x1EF, 0x26B]

ld.lld-11: error: section .header virtual address range overlaps with .bsdata
>>> .header range is [0x1EF, 0x26B]
>>> .bsdata range is [0x205, 0x39E]

ld.lld-11: error: section .bsdata virtual address range overlaps with .entrytext
>>> .bsdata range is [0x205, 0x39E]
>>> .entrytext range is [0x26C, 0x2D3]

ld.lld-11: error: section .text.startup load address range overlaps with .header
>>> .text.startup range is [0x40, 0x204]
>>> .header range is [0x1EF, 0x26B]

ld.lld-11: error: section .header load address range overlaps with .bsdata
>>> .header range is [0x1EF, 0x26B]
>>> .bsdata range is [0x205, 0x39E]

ld.lld-11: error: section .bsdata load address range overlaps with .entrytext
>>> .bsdata range is [0x205, 0x39E]
>>> .entrytext range is [0x26C, 0x2D3]
make[4]: *** [arch/x86/boot/Makefile:108: arch/x86/boot/setup.elf] Error 1
make[3]: *** [arch/x86/Makefile:265: bzImage] Error 2
```

**Comment by dileks** [comment]
Applying `x86 boot compressed runtime relocs` patchset (for example pulling from [1]) solves the issue here.

[1] https://git.kernel.org/pub/scm/linux/kernel/git/tip/tip.git/log/?h=x86/boot

**Comment by dileks** [comment]
Boot on bare metal OK.
```
$ cat /proc/version 
Linux version 5.9.0-rc1-1-amd64-gcc10-lld (sedat.dilek@gmail.com@iniza) (gcc-10 (Debian 10.2.0-5) 10.2.0, LLD 11.0.0 (https://github.com/llvm/llvm-project a450654a52874b094c264e0366c31126c03fdf2d)) #1~bullseye+dileks1 SMP 2020-08-16
```

**Comment by nathanchance** [comment]
This should be fixed as of 5.10-rc1, at least according to my testing. Feel free to reopen if this is not the case.
